private/buf/cmd/buf/buf.go

```diff
@@ -304,6 +305,19 @@ func wrapError(err error) error {
 		return err
 	}
 	if isConnectError {
+		var augmentedConnectError *bufconnect.AugmentedConnectError
+		isAugmentedConnectErr := errors.As(err, &augmentedConnectError)
+		if isPossibleNewCLIOldBSRError(connectErr) && isAugmentedConnectErr {
```

Comment:
What about leaving this under the switch for `connect.CodeUnknown` below? It wasn't clear that `isPossibleNewCLIOldBSRError` mutates the message to me at first.

Reply:
Doesn't mutate the error (unless I'm missing it); it's just introspecting for a particular message that the BSR returns. Felt like this was better handled in one spot so it was clear what we were looking for and we could put the error message there.

I think in the future, when we can drop the `connect.CodeUnknown` check, we could move this to be included in the switch, since it'll only have to check for `CodeUnimplemented`?